scipy/stats/qmc.py

```diff
@@ -0,0 +1,648 @@
+        on the sample and not the theoretical space, unit cube. Thus min and
+        max values of the sample will coincide with the bounds.
+    iterative : bool
+        Must be False if not using it for updating the discrepancy.
+
+    Returns
+    -------
+    discrepancy : float
+        Centered discrepancy.
+
+    References
+    ----------
+    [1] Fang et al. "Design and modeling for computer experiments",
+      Computer Science and Data Analysis Series Science and Data Analysis
+      Series, 2006.
+
+    """
+    sample = np.asarray(sample)
+
+    n_samples, dim = sample.shape
+
+    if iterative:
+        n_samples += 1
+
+    # Sample scaling from bounds to unit hypercube
+    if bounds is not None:
+        min_ = bounds.min(axis=0)
+        max_ = bounds.max(axis=0)
+        sample = (sample - min_) / (max_ - min_)
+
+    abs_ = abs(sample - 0.5)
+    disc1 = np.sum(np.prod(1 + 0.5 * abs_ - 0.5 * abs_ ** 2, axis=1))
+
+    prod_arr = 1
+    for i in range(dim):
+        s0 = sample[:, i]
+        prod_arr *= (1 +
+                     0.5 * abs(s0[:, None] - 0.5) + 0.5 * abs(s0 - 0.5) -
+                     0.5 * abs(s0[:, None] - s0))
+    disc2 = prod_arr.sum()
+
+    c2 = ((13.0 / 12.0) ** dim - 2.0 / n_samples * disc1 +
+          1.0 / (n_samples ** 2) * disc2)
+
+    return c2
+
+
+def update_discrepancy(x_new, sample, initial_disc, bounds=None):
+    """Update the discrepancy with a new sample.
+
+    Parameters
+    ----------
+    x_new : array_like (1, k_vars)
+        The new sample to add in `sample`.
+    sample : array_like (n_samples, k_vars)
+        The initial sample.
+    initial_disc : float
+        Centered discrepancy of the `sample`.
+    bounds : tuple or array_like ([min, k_vars], [max, k_vars])
+        Desired range of transformed data. The transformation apply the bounds
+        on the sample and not the theoretical space, unit cube. Thus min and
+        max values of the sample will coincide with the bounds.
+
+    Returns
+    -------
+    discrepancy : float
+        Centered discrepancy of the sample composed of `x_new` and `sample`.
+
+    """
+    # Sample scaling from bounds to unit hypercube
+    if bounds is not None:
+        min_ = bounds.min(axis=0)
+        max_ = bounds.max(axis=0)
+        sample = (sample - min_) / (max_ - min_)
+        x_new = (x_new - min_) / (max_ - min_)
+
+    n_samples = len(sample) + 1
+    abs_ = abs(x_new - 0.5)
+
+    disc1 = - 2 / n_samples * np.prod(1 + 1 / 2 * abs_ - 1 / 2 * abs_ ** 2)
+    disc2 = 2 / (n_samples ** 2) * np.sum(np.prod(1 + 1 / 2 * abs_ +
+                                                  1 / 2 * abs(sample - 0.5) -
+                                                  1 / 2 * abs(x_new - sample),
+                                                  axis=1))
+    disc3 = 1 / (n_samples ** 2) * np.prod(1 + abs_)
+
+    return initial_disc + disc1 + disc2 + disc3
+
+
+def perturb_discrepancy(sample, i1, i2, k, disc, bounds=None):
```

Comment:
just to be sure: meant to be private right?

Reply:
Yes, I am not sure one would want to use this for something else.